### ISSUE TITLE
PolicyBrain blog? 

### ISSUE THREAD

**Issue by MattHJensen** [OpenSourcePolicyCenter/PolicyBrain#707]
In response to [this comment](https://github.com/OpenSourcePolicyCenter/PolicyBrain/issues/697#issue-264381609) by @martinholmer, @hdoupe said:

> I think this is another example of a quality teaching or best-practices post. Another one that quickly comes to mind is @talumbau 's post on the requests library and posting data through a user interface to a website. I think it would be useful to have these aggregated in one place as an engineer's blog or something like that. Facebook has one that is fun to check out. These blog posts would be created organically from especially informative comments and threads here. So, the cost wouldn't be much higher than writing the comment itself.

@martinholmer replied:

> I think the blog idea is great. See if @MattHJensen agrees and, if so, discuss with him how the blog should be built and maintained.

I think the blog idea sounds good, but I have a couple of questions.

Would the audience only be developers, or would we also target users of the models with content ("how to" posts, etc.)? [Here's an example](https://github.com/open-source-economics/Tax-Calculator/blob/master/taxcalc/reforms/earnings_shifting.py) of something that could be turned into a user-targeted blog post, also by @martinholmer. 

Who would approve /edit the content. Would the 'editor' be centralized or decentralized? 

We [do have](https://github.com/OpenSourcePolicyCenter/blog-posts) the infrastructure to host a blog.

**Comment by martinholmer** [comment]
With respect to PolicyBrain issue #707, could it be that a FAQ document is better suited for this than a blog?

**Comment by MattHJensen** [comment]
> With respect to PolicyBrain issue #707, could it be that a FAQ document is better suited for this than a blog?

I was thinking that it would be easier to disseminate each concept individually in a blog, but I suppose we could disseminate new additions to a FAQ individually as well. (We could even disseminate new additions to a FAQ in blog form if we wanted to.)

**Comment by martinholmer** [comment]
@MattHJensen said:
> I was thinking that it would be easier to disseminate each concept individually in a blog, but I suppose we could disseminate new additions to a FAQ individually as well. (We could even disseminate new additions to a FAQ in blog form if we wanted to.)

Not sure what you mean by "disseminate".

A FAQ document typically just answers each question, but if the answer is long the FAQ document simply provides a link to the document with the longer answer.  In this way, there could be a central place (that could grow over time a more FAQ are added) that provides links to issue or pull request comments that are deemed to be "important" or "helpful".

@hdoupe

**Comment by MattHJensen** [comment]
> Not sure what you mean by "disseminate".

I mean that it is nice to be able to 'push' content out to people on a regular basis to keep them interested in our work and keep it on their mind. One way to think about this is that it would be nice to have more content to put in regular OSPC, AEI newsletters, and perhaps in the future, project-specific newsletters. 

Blog posts are natural for this, which is why I was initially drawn to the blog idea, but I think we could also, for instance, highlight a new FAQ Q&A in a newsletter.

**Comment by hdoupe** [comment]
@MattHJensen asked a couple questions:

> Would the audience only be developers, or would we also target users of the models with content ("how to" posts, etc.)? Here's an example of something that could be turned into a user-targeted blog post, also by @martinholmer.

The types of audience that I have in mind are: 
1. Open-Source contributors--People who may be interested in our organizational approach.  For example, more people might be interested in learning about our Release Process.  Perhaps, that's a good model for other projects.
2. Python developers--People who are trying to building something in Python.  For example, I haven't found a good, up-to-date blog post on how to set up a Pelican blog.
3. Policy analysts--People who want to see more in-depth explanations of aspects of our models or interesting use cases like @martinholmer's [case](https://github.com/open-source-economics/Tax-Calculator/blob/master/taxcalc/reforms/earnings_shifting.py).

> Who would approve /edit the content. Would the 'editor' be centralized or decentralized?

I have never run a blog, and I definitely have not run one with multiple contributors.  So, I'm not sure how this works.  My inclination is that it would be similar to the code contribution approval process.  Someone submits a blog post.  People who are interested in the topic read it and give critiques.  Then, the contributor revises.  We repeat this process until a consensus is reached.  At that point, someone with write-access to the blog merges the post and pushes to the server.

To summarize, I think that everyone should be able to edit and critique, but there should only be one or two people with the final say.

**Comment by hdoupe** [comment]
@martinholmer asked:
> With respect to PolicyBrain issue #707, could it be that a FAQ document is better suited for this than a blog?

A FAQ would be useful, but I think it would be a complement to a blog not a substitute.  Some questions in a FAQ may point to a blog post for a longer answer.  Some blog-posts may not work well in the FAQ format.  Some items in a FAQ are not long enough to work well in a blog format.

**Comment by MattHJensen** [comment]
@hdoupe said: 

> The types of audience that I have in mind are:
> 
> 1. Open-Source contributors--People who may be interested in our organizational approach. For example, more people might be interested in learning about our Release Process. Perhaps, that's a good model for other projects.
> 2. Python developers--People who are trying to building something in Python. For example, I haven't found a good, up-to-date blog post on how to set up a Pelican blog.
> 3. Policy analysts--People who want to see more in-depth explanations of aspects of our models or interesting use cases like @martinholmer's case.

This makes sense to me, although I'd suggest segmenting the policy analysts into technical (PolicyBrain) and non-technical (upstream project). We can tag content based on user. In the policy analyst content, we'll need to be _very_ careful to focus on the "how to" side of things rather than the results themselves. 

> I have never run a blog, and I definitely have not run one with multiple contributors. So, I'm not sure how this works. My inclination is that it would be similar to the code contribution approval process. Someone submits a blog post. People who are interested in the topic read it and give critiques. Then, the contributor revises. We repeat this process until a consensus is reached. At that point, someone with write-access to the blog merges the post and pushes to the server.
> 
> To summarize, I think that everyone should be able to edit and critique, but there should only be one or two people with the final say.

This makes sense to me as well. We'll likely need a different "final say" person depending on the type of content/audience.

**Comment by GoFroggyRun** [comment]
[Github pages](https://pages.github.com/) might be a good place to start with. Moving some (all) of the markdown files we currently have to the potential blog page is desirable in my opinion.

**Comment by hdoupe** [comment]
@MattHJensen said:

> In the policy analyst content, we'll need to be very careful to focus on the "how to" side of things rather than the results themselves.

Right, that makes sense. Focusing on the "how to" side is important. Policy analysis blog posts should focus primarily on the tools that are used in my opinion.

> This makes sense to me as well. We'll likely need a different "final say" person depending on the type of content/audience.

I agree. To get started, I think that there should be a "final say" person for software engineering type content and a person for blogs that are demonstrating how to use software tools to analyze a policy. 

@andersonfrailey wrote a [blog post](https://andersonfrailey.github.io/tcja-viz.html) describing how he built a cool TCJA visualization. It only covers the methods for building the plot and does not discuss the results in any way. I think it's a good example of a technical post. He also might be interested in cross-posting onto the OSPC blog.

**Comment by andersonfrailey** [comment]
I'd be happy to cross-post. I'll want to update it to use whatever the latest version of Tax-Calculator is at the time of posting first though.
